Sources/Internals/PassthroughView.swift

```diff
@@ -76,19 +76,21 @@ final class PassthroughView: UIView {
 	}
 
 	private func updatePositioning() {
-		var frame = superview?.bounds ?? .zero
-
-		if let superview = superview as? UIScrollView {
-			if #available(iOS 11.0, *) {
-				frame = frame.inset(by: superview.safeAreaInsets)
-			} else {
-				frame = frame.inset(by: superview.contentInset)
+		DispatchQueue.main.async {
+			var frame = superview?.bounds ?? .zero
+
+			if let superview = superview as? UIScrollView {
+				if #available(iOS 11.0, *) {
+					frame = frame.inset(by: superview.safeAreaInsets)
+				} else {
+					frame = frame.inset(by: superview.contentInset)
+				}
 			}
-		}
 
-		frame.origin = .zero
```

Comment:
I'm not sure what the reason was that this was added. If any specific reason, I'd love to hear it.

`inset(by:)` will update `size` and `origin` according to the `safeAreaInsets`. Manually setting the frame to `.zero` afterwards results in all horizontal insets (left + right) to be applied on the right side only. We can see this behaviour whenever there is a safe area on the left side: the view will not be positioned correctly.

Reply:
I think this was to handle some UIKit craziness on maybe older systems, where the stateful view would be pushed down in some cases. This should be more extensively tested in some apps to ensure we're not breaking a bunch of them.

Might be better to create a separate PR for this?

Reply:
This already causes an issue in the apps where this code is currently used.
Whenever there is a safe area combined with rotating the device (eg: rotating an iPhone towards landscape results in an incorrect positioning of the state view).
Perhaps we can define a list apps/use cases we should test?

Reply:
TVL, Eevee, Cocom, Trenara, ... I think we should just mention it when we create a release. We can always revert it if need be, or add more logic 🤷